modules/multitile_support/src/tile_map.xc

```diff
@@ -4,10 +4,170 @@
 #include "platform.h"
 
 extern "C" {
+
+
+#ifndef MINIMISE_MAIN_TILE_ARGS 
+
+
+
 void main_tile0(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile1(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile2(chanend c0, chanend c1, chanend c2, chanend c3);
 void main_tile3(chanend c0, chanend c1, chanend c2, chanend c3);
+
+
+#else // MINIMISE_MAIN_TILE_ARGS defined
+
+
+
+// -------------------- Tile 0 prototype ------------------------------
+#if ((PLATFORM_SUPPORTS_TILE_0 == 1) && (PLATFORM_USES_TILE_0 == 1))
+                      void main_tile0(
+  
+// c0  
+// channel c0  is loopback so it is never in the minimised argument set
+
+
+// c1
+#if ((PLATFORM_SUPPORTS_TILE_1 == 1) && (PLATFORM_USES_TILE_1 == 1))
+                      chanend c1
+#define ARGS_STARTED
+#endif
+
+
+// c2
+#if ((PLATFORM_SUPPORTS_TILE_2 == 1) && (PLATFORM_USES_TILE_2 == 1))
+#ifdef ARGS_STARTED
+                      ,  
+#endif
+                      chanend c2
+#define ARGS_STARTED
+#endif
+
+
+// c3
+#if ((PLATFORM_SUPPORTS_TILE_3 == 1) && (PLATFORM_USES_TILE_3 == 1))
+#ifdef ARGS_STARTED
```

Comment:
Functions whose prototype changes based on macro definitions cause confusion IMO. You need to run the preprocessor on this code before you can know what the function you need to define is. 

Open to being persuaded otherwise if there are reasons behind doing it this way

Reply:
I would usually agree on this point and I agree it's not a perfect solution. However, I put in two mitigations:
1. I added text to the README describing what I believe will be the most common setup (a two-tile system). The README is ordered so that users wanting the simple prototypes get the prototypes and all the information they need to configure the build to use those prototypes. 
2. The proposed changes are guarded by the define MINIMISE_MAIN_TILE_ARGS. If this is not set the prototypes are unchanged.